### DISCUSSION TITLE
Rendering the same page for multiple paths

### DISCUSSION THREAD

**Discussion by mithi** [vercel/next.js#20254] [Help] [answered]
Hi, 

say I have a `notebooks/math/` with pages like 
`notebook/math/1`, `notebook/math/2` say up to `notebook/math/10` which is the last page.
and I have used static site generation (ssg) having static paths successfully generate these pages. 

But say I want the routes like:
- `notebook/math?page=1`, 
- `notebook/math?page=0`, 
- `notebook/math?page=-1`, 
- `notebook/math/-1`
- `notebook/math/abcsdaljdsa`  

to render the same page as `notebook/math/1`.

Additionally `notebook/math/10000`, `notebook/math?page=10000` should render `notebook/math/10` or whatever the last page is. 

Is there a way to elegantly do this? 
I've seen https://nextjs.org/docs/api-reference/next.config.js/redirects but it doesn't handle all cases and it redirects instead of rendering the same thing given different routes. 

Thanks!

**Comment by ShubhamVerma1811** [comment]
From what I understand, this can help you acheive that.
https://nextjs.org/docs/routing/dynamic-routes

**Reply by mithi** [reply]
I have referred to that page as I was implementing my dynamic routes.

A variation of the code below works when I don't do static site generation
```js
import { useRouter } from 'next/router'

const Post = () => {
  const router = useRouter()
  const { pid } = router.query

  return <p>Post: {pid}</p>
}
```

But when I do use `getStaticPaths` and `getStaticProps`, and I do [fallback: true](https://nextjs.org/docs/basic-features/data-fetching#fallback-pages) it will just produce an error. If I do `fallback: false` it will just render a 404 page as expected. To give more context, I'll elaborate the things I have tried and the corresponding outcomes as much as I can in my next comment.

**Comment by mithi** [comment, marked as answer]
1. Create 
- `pages/notebook/[section].js`
- `pages/notebook/[section]/[page].js` 
- `pages/notebook/page/[[...others]].js`

2. In `pages/notebook/[section]/[page].js`, use `getStaticProps` and `getStaticPaths` as you would normally do.

3. In both `pages/notebook/page/[[...others]].js` and `pages/notebook/[section].js`, be sure to use `getServerSideProps`. If you don't, you won't be able to access the query. See also [Advanced Features: Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization)

```js
import { useRouter } from "next/router"

export async function getServerSideProps() {
    return {
        props: {}, // will be passed to the page component as props
    }
}

const Home = () => {
    const router = useRouter()
    console.log(router) // you'll be able to access the pathname and, the 'other' array, other query parameters here
    return <div>...</div>
}

export default Home
```
